src/messages/mod.rs

```diff
@@ -237,14 +275,77 @@ impl Debug for SecurityMetadata {
 }
 
 /// Wrapper around a routing message, signed by the originator of the message.
-#[derive(Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
+#[derive(Debug, Eq, PartialEq, Clone, Hash)]
+pub struct PeekedSignedRoutingMessageInfo {
+    /// Destination location
+    pub dst: Location,
+}
+
+impl<'de> Deserialize<'de> for PeekedSignedRoutingMessageInfo {
```

Comment:
I wonder, can't we just do:
```rust
let peeked_message: PeekedSignedRoutingMessageInfo = deserialize(&bytes)?;
```
Or would that fail because of not consuming the bytes to the end?

Reply:
We deserialize a tupple, since we serialize a tuple.
But I'll look whether I can just deserialize to a single `Location` value.

Cannot deserialize(&bytes)?; because bytes contain an enum of either direct or hop

Reply:
I think you should be able to serialize a `Message` but deserialize `PeekMessage`. Then you wouldn't need this custom deserialization logic. But I'm not sure it would really work because it would not consume the whole buffer. But might be worth a try.

Reply:
We already not consuming the full buffer, so not really an issue. as I said will try.

Reply:
Well that worked, that was an unfortunate rabbit hole.